### ISSUE TITLE
compress/flate: cleanup huffman_bit_writer.go

### ISSUE THREAD

**Issue by dsnet** [golang/go#18458]
The [`huffman_bit_writer.go` file contains references to some "extended window"](https://github.com/golang/go/blob/c1cd64d0ac2001ec8fe6f253e4e95561a444d533/src/compress/flate/huffman_bit_writer.go#L69-L73) implementation, which seems to provide the ability to reference a distance greater than 15 bits (up to 22 bits it seems). This goes contrary to RFC 1951, which distinctly says the maximum distance is 32768. As far as I can tell, this behavior is disabled since `logWindowSize = 15`, which ensures that we will never use a distance greater than this. Even if it were somehow usable, we should fix this since that means we could accidentally produce non-RFC1951 compliant files.

This logic has been included since 2009, so I am unsure as to what the original purpose of this is. Can we delete these?

\cc @nigeltao @krasin2

**Comment by krasin** [comment]
More active members of the Go team might correct me, but my guess is it's okay to remove. Note: I have not touched the Go stdlib for years.

**Comment by minux** [comment]
This makes me wonder whether we can add a special
mode to cmd/cover to track coverage (usage) of such
data tables.

I have encountered quite some cases where I want to
see if my tests forced the code to use all the entries of
a map or an array/slice, and each time I wrote ad-hoc
and case-specific code. It will be very cool if there are
some standard tool that could help with this task.

Of course, the hard part is knowing which variables to
track and figure out the story for dynamically generated
objects.
